src/rgw/rgw_process.h

```diff
@@ -38,6 +38,8 @@ struct RGWProcessEnv {
   int port;
   std::string uri_prefix;
   std::shared_ptr<rgw::auth::StrategyRegistry> auth_registry;
+  //maybe there is a better place to store the rate limit data structure
+  ActiveRateLimiter* ratelimiting;
```

Comment:
if the owner is a local variable in rgw_main then this is fine to leave it as a bare pointer here

Reply:
if the owner is a local variable in rgw_main then this is fine to leave it as a bare pointer here